app/javascript/App/bundles/NewsfeedPage/NewsfeedPage.tsx

```diff
@@ -65,87 +64,34 @@ class NewsfeedPage extends React.Component<Props, State> {
   }
 
   componentDidMount() {
+    window.addEventListener('resize', debounce(() => this.forceUpdate(), 500))
+
     if (this.getContentType() === "coin") {
-        this.fetchNewsItemsForCoin(this.props.coinSlug);
+        this.props.fetchNewsItemsForCoin(this.props.coinSlug)
     } else {
-      this.fetchAllNewsItems()
+      if (this.props.newslist.length > 0) {
+        return;
+      }
+      this.props.fetchAllNewsItems();
     }
   }
 
   componentDidUpdate(prevProps, prevState, snapshot) {
-    window.addEventListener('resize', debounce(() => this.forceUpdate(), 500))
-
     if (this.props.coinSlug !== prevProps.coinSlug) {
-      this.setState(
-        {
-          status: STATUSES.LOADING,
-        },
-        () =>
-          this.props.coinSlug === undefined
-            ? this.fetchAllNewsItems()
-            : this.fetchNewsItemsForCoin(this.props.coinSlug),
-      )
+      this.props.fetchNewsItemsForCoin(this.props.coinSlug)
     }
   }
 
   closeTips = () => {
     this.setState({ initialRenderTips: false })
   }
 
-  getNewsItem(): NewsItem | undefined {
-    return _.find(this.state.sortedNewsItems, ['id', parseInt(this.props.newsItemId)]);
-  }
-
-  getCoinInfo(): Coin {
-    return _.find(this.props.coinList, ['slug', this.props.coinSlug]);
-  }
-
   static getDerivedStateFromProps(props, state) {
-    const { match, location, history } = props
-    if (state.coinSlug === match.params.coinSlug 
-      && state.newsItemId === props.newsItemId) {
-      return null
-    } else {
-      return { coinSlug: match.params.coinSlug, newsItemId: props.newsItemId }
+    return {
+      coinInfo: _.find(props.coinlist, ['slug', props.coinSlug]) 
```

Comment:
Why do you need this in state?

Reply:
Previously there wasn't api action to get the `coinWithDetails` by `slug`. So it was needed to pass `coinID` to `CoinBody` to get full info. Now it's all deleted and added new action `by_slug` to get `coinWithDetails` directly from `CoinBody`. Much simpler